src/peft/utils/pissa_utils.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Reference paper: https://arxiv.org/abs/2404.02948
+
+from __future__ import annotations
+
+
+import os
+import torch
+from safetensors import safe_open
+from safetensors.torch import save_file
+import json
+
+
+def pissa_pre_training_saving(peft_model, tokenizer, save_path, push_to_hub=None):
```

Comment:
Let's add a docstring. Also, let's add `**kwargs` to the signature and pass those to `save_pretrained` below.